master/pkg/searcher/asha.go

```diff
@@ -258,7 +259,11 @@ func (s *asyncHalvingSearch) trialExitedEarly(
 		highestRungIndex := s.trialRungs[requestID]
 		for rungIndex := 0; rungIndex <= highestRungIndex; rungIndex++ {
 			rung := s.rungs[rungIndex]
-			rung.metrics = nil
+			for i, trialMetric := range rung.metrics {
+				if trialMetric.requestID == requestID {
+					rung.metrics = append(rung.metrics[:i], rung.metrics[i+1:]...)
```

Comment:
I think we can `break` after this right, there should only be one metric per trial per rung.